lib/vector/Vlib/color_read.c

```diff
@@ -49,7 +49,9 @@ int Vect_read_colors(const char *name, const char *mapset,
     if (colors)
         Rast_init_colors(colors);
 
-    strcpy(xname, name);
+    if (G_strlcpy(xname, name, sizeof(xname)) >= sizeof(xname)) {
+        G_fatal_error(_("Vector map name <%s> is too long"), name);
```

Comment:
As the function do have a return value of failure (and is part of the GRASS library), let us issue a warning instead and return `-1`.

Reply:
Done